src/dropdowns/dropdown.ts

```diff
@@ -0,0 +1,209 @@
+import {Directive, Input, Output, EventEmitter, HostBinding, HostListener, ElementRef, OnInit, AfterContentInit, OnDestroy, ContentChildren, QueryList} from 'angular2/core';
+import {NglDropdownTrigger} from './dropdown-trigger';
+import {NglDropdownItem} from './dropdown-item';
+import {NglPageClickService} from './page-click.service';
+import {toBoolean} from '../util/util';
+
+const openEventEmitter: EventEmitter<any> = new EventEmitter<any>(false);
+
+@Directive({
+  selector: '[nglDropdown]',
+  host: {
+    'class': 'slds-dropdown-trigger--click',
+    'tabindex': '0',
+  },
+  providers: [NglPageClickService],
+})
+
+export class NglDropdown implements OnInit, AfterContentInit, OnDestroy {
+
+  @Input('nglDropdown') set values(values: any[]) {
+    this._values = values;
+    this._updateItems(this._values || []);
+  }
+  get values() {
+    return this._values;
+  }
+  @Input('open') set isOpen(isOpen: boolean | string) {
+    isOpen = toBoolean(isOpen);
+     if (isOpen) {
+       this.handlePageClickEvents = false;
+       setTimeout(() => this.handlePageClickEvents = true);
+     }
+    //setTimeout(() => {
+    this._isOpen = <boolean>isOpen;
+    if (isOpen) {
+      setTimeout(() => {
+        this.focusNextItem();
+      });
+    }
+    //});
+  }
+  get isOpen() {
+    return this._isOpen;
+  }
+  @Input() handlePageEvents = true;
+  @ContentChildren(NglDropdownTrigger) triggers: QueryList<NglDropdownTrigger> = new QueryList<NglDropdownTrigger>();
+  @ContentChildren(NglDropdownItem) items: QueryList<NglDropdownItem> = new QueryList<NglDropdownItem>();
+  @Output('openChange') isOpenChange: EventEmitter<boolean> = new EventEmitter<any>(false);
+  @Output('select') selectionEmitter: EventEmitter<any> = new EventEmitter<any>(false);
+  @HostBinding('class.slds-is-open')
+  @HostBinding('attr.aria-expanded')
+  get __isOpen() {
+    return this.isOpen;
+  }
+
+  private handlePageClickEvents = true;
+  private _isOpen = false;
+  private _values: any[] = [];
+  private openEventSubscription: any;
+  private clickEventSubscription: any;
+
+  @HostListener('keydown.esc', ['$event'])
+  @HostListener('keydown.tab')
+  onKeydownClose($event: Event) {
+    this.toggle(false);
+    if ($event) {
+      setTimeout(() => {
+        if (this.triggers.length) {
+          this.triggers.first.focusSelf();
+        }
+      });
+    }
+  }
+  @HostListener('keydown.arrowdown', ['$event'])
+  @HostListener('keydown.arrowright', ['$event'])
+  onKeydownFocusNext($event: Event) {
+    $event.preventDefault();
+    this.focusNextItem();
+  }
+  @HostListener('keydown.arrowup', ['$event'])
+  @HostListener('keydown.arrowleft', ['$event'])
+  onKeydownFocusPrevious($event: Event) {
+    $event.preventDefault();
+    this.focusPreviousItem();
+  }
```

Comment:
If you want you can combine all the four `@HostListener('keydown.*)` in one function and the call the appropriate function based on `$event.keyIdentifier' or `$event.code'`. It's up to you if you want to make this change :)